### ISSUE TITLE
Gradients of outputs w.r.t inputs during eval

### ISSUE THREAD

**Issue by ezgicakirofficial** [mit-han-lab/torchsparse#70]
Hello, 
I am new on torchsparse library and point cloud data. 
I would like to get gradients of outputs with respect to inputs. Normally, in pytorch, we should be able to get it by torch.autograd.grad(..) function [https://pytorch.org/docs/stable/autograd.html#torch.autograd.grad] but this does not work for SparseTensor. I try to pass input.F as inputs parameter of this function, it did not work. It returns None.

1- So, how can I get them? I looked up #6 , but could not managed still. It return None all the time.
One thing: I try to get them after model.eval(). Is this the problem? But I want to get it during eval. Is there any other way to calculate gradients?

Thank you in advance.

**Comment by zhijian-liu** [comment]
You probably need to first set `inputs.F.requires_grad` to `True`. I usually get the gradient of input by accessing `inputs.F.grad` after doing `loss.backward()`.

**Comment by ezgicakirofficial** [comment]
I did what you have described. Here is the code in summary: 
`
model.eval()
outputs = model(inputs)
inputs.F.requires_grad = True
loss = outputs.sum()
loss.backward()
print(inputs.F.grad)
`

And it print None.. Is it wrong? 
I tried this as well and the result is the same:

`
model.eval()
outputs = model(inputs)
inputs.F.requires_grad = True
loss = criterion(outputs, targets)
optimizer.zero_grad()
loss.backward()
print(inputs.F.grad)
`
 
What do I miss? :(

**Comment by zhijian-liu** [comment]
Could you please try that in the training mode to see whether it's the reason?

**Comment by zhijian-liu** [comment]
Btw, you need to put `inputs.F.requires_grad = True` to be before the model inference.

**Comment by ezgicakirofficial** [comment]
Aah, yes, I figured it out after many trials. :) Thank you for your help!
